pkg/alexa/request.go

```diff
@@ -88,16 +135,82 @@ type SlotValue struct {
 	Resolutions *Resolutions `json:"resolutions"`
 }
 
+// Slots returns the list of slots, an empty list if no intent was found
+func (r *RequestEnvelope) Slots() map[string]*Slot {
+	i, err := r.Intent()
+	if err != nil {
+		return map[string]*Slot{}
+	}
+
+	if i.Slots == nil {
+		return map[string]*Slot{}
+	}
+
+	return i.Slots
+}
+
+// Slot returns the named slot or an error
+func (r *RequestEnvelope) Slot(name string) (Slot, error) {
+	i, err := r.Intent()
+	if err != nil {
+		return Slot{}, err
+	}
+
+	s, ok := i.Slots[name]
+	if !ok {
+		return Slot{}, ErrorNoSlot
```

Comment:
Perhaps a more general `ErrNotFound` would be easier to deal with for all of the cases you added. If you want to record what was not found in the different cases, you can make an error type `NotFoundError` that has a string for the type, like `slot`.

Reply:
still working on it 😀 the golangci-lint still reports some errors that I have to take care of...